diff --git a/src/vos/vos_common.c b/src/vos/vos_common.c
index db0e7243f..0a986793a 100644
--- a/src/vos/vos_common.c
+++ b/src/vos/vos_common.c
@@ -40,6 +40,10 @@ static pthread_mutex_t	mutex = PTHREAD_MUTEX_INITIALIZER;
 static bool vsa_nvme_init;
 struct vos_tls	*standalone_tls;
 
+#ifdef VOS_STANDALONE
+unsigned int tmp_count;
+#endif
+
 struct vos_tls *
 vos_tls_get(void)
 {
diff --git a/src/vos/vos_internal.h b/src/vos/vos_internal.h
index cb0cbd688..469870e35 100644
--- a/src/vos/vos_internal.h
+++ b/src/vos/vos_internal.h
@@ -243,7 +243,7 @@ struct vos_dtx_act_ent {
 
 extern struct vos_tls	*standalone_tls;
 #ifdef VOS_STANDALONE
-unsigned int tmp_count;
+extern unsigned int tmp_count;
 #define VOS_TIME_START(start, op)		\
 do {						\
 	if (standalone_tls->vtl_dp == NULL)	\
