t a/SConstruct b/SConstruct
index d2b2a10e..a7dce604 100644
--- a/SConstruct
+++ b/SConstruct
@@ -47,7 +47,7 @@ def preload_prereqs(prereqs):
     prereqs.define('readline', libs=['readline', 'history'],
                    package='readline')
     reqs = ['cart', 'argobots', 'pmdk', 'cmocka',
-            'uuid', 'crypto', 'fuse', 'protobufc']
+            'uuid', 'crypto', 'fuse', 'protobufc', 'hwloc', 'ompi']
     if not is_platform_arm():
         reqs.extend(['spdk', 'isal'])
     prereqs.load_definitions(prebuild=reqs)
