diff --git a/SConstruct b/SConstruct
index 22f6ff4..d937d76 100644
--- a/SConstruct
+++ b/SConstruct
@@ -73,7 +73,6 @@ def run_checks(env):
     """Run all configure time checks"""
 
     cenv = env.Clone()
-    cenv.Append(CFLAGS='-Werror')
     config = Configure(cenv)
 
     if config.CheckHeader('stdatomic.h'):
@@ -105,7 +104,7 @@ def scons():
         env.Append(SHLIBSUFFIX='.so')
 
     # Compiler options
-    env.Append(CCFLAGS=['-g3', '-Wshadow', '-Wall', '-Werror', '-fpic',
+    env.Append(CCFLAGS=['-g3', '-Wshadow', '-Wall', '-fpic',
                         '-D_GNU_SOURCE', '-DD_LOG_V2'])
     env.Append(CCFLAGS=['-O2', '-pthread'])
     env.Append(CFLAGS=['-std=gnu99'])
