diff --git a/src/cart/crt_group.c b/src/cart/crt_group.c
index 150d7e0..06c2fcc 100644
--- a/src/cart/crt_group.c
+++ b/src/cart/crt_group.c
@@ -3293,7 +3293,7 @@ crt_grp_config_psr_load(struct crt_grp_priv *grp_priv, d_rank_t psr_rank)
 		D_GOTO(out, rc = d_errno2der(errno));
 	}
 
-	D_ALLOC(grpname, CRT_GROUP_ID_MAX_LEN);
+	D_ALLOC(grpname, CRT_GROUP_ID_MAX_LEN + 1);
 	if (grpname == NULL)
 		D_GOTO(out, rc = -DER_NOMEM);
 	snprintf(fmt, 64, "%%*s%%%ds", CRT_GROUP_ID_MAX_LEN);
@@ -3338,7 +3338,7 @@ crt_grp_config_psr_load(struct crt_grp_priv *grp_priv, d_rank_t psr_rank)
 		D_GOTO(out, rc = -DER_INVAL);
 	}
 
-	D_ALLOC(addr_str, CRT_ADDR_STR_MAX_LEN);
+	D_ALLOC(addr_str, CRT_ADDR_STR_MAX_LEN + 1);
 	if (addr_str == NULL)
 		D_GOTO(out, rc = -DER_NOMEM);
 
